### ISSUE TITLE
Object-alt rule should ignore objects that don't load things

### ISSUE THREAD

**Issue by WilcoFiers** [dequelabs/axe-core#3656]
The following is from an [ACT inapplicable example](https://wai-wcag-act-rules.netlify.app/standards-guidelines/act/rules/8fc3b6/proposed/#inapplicable-example-8), axe shouldn't fail this:

```html
<object data="/invalid/url/index.html">
  <img src="/WAI/content-assets/wcag-act-rules/test-assets/shared/w3c-logo.png" alt="W3C logo" />
</object>
```

I'm not sure how to directly detect whether an object loaded something, but detecting that its pass-through content is displayed can easily be done by just checking that it's children have a position, for example by calling `getClientRects()` on them. There are probably other ways to do it too.

To do this we'll probably need a new matches method for objects that can filter out objects which show their content.

**Comment by padmavemulapati** [comment]
Validated with the latest axe-core develop branch code base, 
for the test script:
```
<object data="/invalid/url/index.html">
  <img src="/WAI/content-assets/wcag-act-rules/test-assets/shared/w3c-logo.png" alt="W3C logo" />
</object>
```
`object-alt` rule is appearing in `inapplicable` , not failing


![Image](https://user-images.githubusercontent.com/47409579/195615460-ed8c8601-8b2f-4e3b-b64c-4951b3136fc3.png)
